easybuild/tools/toolchain/compiler.py

```diff
@@ -231,9 +231,19 @@ def _set_compiler_flags(self):
         fflags = [self.options.option(x) for x in self.COMPILER_F_FLAGS + self.COMPILER_F_UNIQUE_FLAGS \
                   if self.options.get(x, False)]
 
+        # Allow a user-defined default optimisation
+        if build_option('default_optimisation'):
+            if build_option('default_optimisation') in self.COMPILER_OPT_FLAGS:
+                default_opt = self.options.option(build_option('default_optimisation'))
```

Comment:
you can simplify this a bit by only calling out to `build_option` once:

```python
default_opt_level = build_option('default_opt_level')
if default_opt_level in self.COMPILER_OPT_FLAGS:
    default_opt = default_opt_level
else:
    raise ...
```

Note that the `default_opt` default setting has just moved into `options.py` (see my other remark), no need to hardcode it anymore here